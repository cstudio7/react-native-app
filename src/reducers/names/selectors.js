export const getNames = (state, { tab }) => {
  return state.names[tab];
};
