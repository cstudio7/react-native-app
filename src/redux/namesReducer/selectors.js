export const filterFavorites = names => {
  const filter = genderNames => genderNames.filter(name => name.isFavorite);
  const female = filter(names.female);
  const male = filter(names.male);
  return { female, male };
};
