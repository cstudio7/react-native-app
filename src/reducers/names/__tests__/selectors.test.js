import { getNames } from '../selectors';

const names = Object.freeze({
  female: [{ name: 'sveta', isFavorite: true }],
  male: [{ name: 'artur', isFavorite: true }, { name: 'pierre' }]
});

const state = Object.freeze({
  names
});

it('returns female names', () => {
  const props = {
    tab: 'female'
  };
  expect(getNames(state, props)).toMatchSnapshot();
});

it('returns male names', () => {
  const props = {
    tab: 'male'
  };
  expect(getNames(state, props)).toMatchSnapshot();
});
