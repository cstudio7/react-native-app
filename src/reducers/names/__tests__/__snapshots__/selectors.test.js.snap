// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns female names 1`] = `
Array [
  Object {
    "isFavorite": true,
    "name": "sveta",
  },
]
`;

exports[`returns male names 1`] = `
Array [
  Object {
    "isFavorite": true,
    "name": "artur",
  },
  Object {
    "name": "pierre",
  },
]
`;
